src/Runner.Worker/Handlers/NodeScriptActionHandler.cs

```diff
@@ -134,6 +134,29 @@ public async Task RunAsync(ActionRunStage stage)
             // Remove environment variable that may cause conflicts with the node within the runner.
             Environment.Remove("NODE_ICU_DATA"); // https://github.com/actions/runner/issues/795
 
+            if(Data.NodeVersion == Constants.Runner.DeprecatedNodeVersion && (ExecutionContext.Global.Variables.GetBoolean(Constants.Runner.Features.Node16Warning) ?? false))
+            {
+                var repoAction = Action as RepositoryPathReference;
+                var warningActions = new HashSet<string>();
+                if (ExecutionContext.Global.Variables.TryGetValue(Constants.Runner.DeprecatedNodeDetectedAfterEndOfLifeActions, out var deprecatedNodeWarnings))
+                {
+                    warningActions = StringUtil.ConvertFromJson<HashSet<string>>(deprecatedNodeWarnings);
+                }
+
+                var repoActionFullName = "";
```

Comment:
I'd propose below code, it's a bit simpler and reduces amount of mutable vars. Plus repoActionFullName is used only in this if-else block anyway.
```suggestion
                 if (string.IsNullOrEmpty(repoAction.Name))
                {
                  // local actions don't have a 'Name'
                   warningActions.Add(repoAction.Path);
                }
                else
                {
                     warningActions.Add($"{repoAction.Name}/{repoAction.Path ?? string.Empty}".TrimEnd('/') + $"@{repoAction.Ref}");
                }
```

Reply:
Thanks for you comment. Applied 👍